product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -10,6 +10,38 @@
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
+    @api.multi
+    def name_get(self):
+        res = super(ProductProduct, self.with_context(customerinfo=True)).\
+            name_get()
+        return res
+
+    @api.model
+    def _name_search(self, name='', args=None, operator='ilike', limit=80):
```

Comment:
Why this method doesn't follow a similar approach than the other?

Reply:
because `name_get` uses `search`, but `_name_search` uses `_search` instead. Thus, it means would need to also hack `_search`, but then my head would explode, trying to making compatible the hack of `search` with the hack of `_search` hahaha

Reply:
`_search` returns ids, so here I cannot do same hack as done in `search`.